### ISSUE TITLE
ALERename fails on double renaming

### ISSUE THREAD

**Issue by bogdan** [dense-analysis/ale#3642]
## Information

**VIM version**

NVIM v0.4.3
Build type: Release

Operating System:  10.15.5

## What went wrong

ALERename fails on double renaming with typescript and tsserver

## Reproducing the bug

Suppose you have  2 files:

``` typescript
// ./one.ts
export default class One {
}
// ./two.ts
import One from './a';

class Two extends One {
}
```

1. Open `one.ts`.
2. Put the cursor on `One` class name.
3. Call `:ALERename`
4. Rename to `One1`
5. Call `:ALERename`
6. Rename to `One11`
7. Open `two.ts` file.

Expected `Two` to extend `One11`  but it is extending `One1` instead.

### :ALEInfo

https://gist.github.com/bogdan/cd13d82cd808b844f99c1695aeb3974a

**Comment by w0rp** [comment]
Potentially tricky to fix, but definitely worth fixing.

**Comment by jeremija** [comment]
@bogdan I think I was able to come up with a proof of concept that fixes this issue: the solution is to send the `textDocument/didChange` message after modifying the files. You can have a look at the changes here:

https://github.com/jeremija/ale/commit/db6b7d1c0acf6cb1af19b6e278deb16a7e984830

I just don't have the time to fix all the broken tests and submit a complete MR at the moment, but if anyone want's to do it - feel free!

**Comment by hsanson** [comment]
Fixed with #4049
